python/lsst/sconsUtils/scripts.py

```diff
@@ -481,9 +507,31 @@ def s(l):
             swigMods.extend(
                 state.env.SwigLoadableModule("_" + name, src, LIBS=state.env.getLibs("main python"))
             )
+
+        # Warn about insisting that a test in pySingles starts with test_ and therefore
+        # might be automatically discovered by pytest. This warning could only be issued
```

Comment:
To answer the functional question, I'm in favor of always generating the warning to make sure the developer aware of the potential problem.